assets/src/dashboard/app/api/useSettingsApi.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useReducer } from 'react';
+import queryString from 'query-string';
+
+/**
+ * Internal dependencies
+ */
+import settingsReducer, {
+  defaultSettingsState,
+  ACTION_TYPES as SETTINGS_ACTION_TYPES,
+} from '../reducer/settings';
+
+export default function useSettingsApi(
+  dataAdapter,
+  { globalStoriesSettingsApi }
+) {
+  const [state, dispatch] = useReducer(settingsReducer, defaultSettingsState);
+
+  const fetchSettings = useCallback(async () => {
+    if (!globalStoriesSettingsApi) {
+      dispatch({
+        type: SETTINGS_ACTION_TYPES.FETCH_SETTINGS_FAILURE,
+        payload: {
+          message: {
+            body: __('Cannot connect to data source', 'web-stories'),
+            title: __('Unable to find settings data', 'web-stories'),
+          },
+        },
+      });
+    }
+    try {
+      const response = await dataAdapter.get(
+        queryString.stringifyUrl({
+          url: globalStoriesSettingsApi,
+        })
+      );
+
+      dispatch({
+        type: SETTINGS_ACTION_TYPES.FETCH_SETTINGS_SUCCESS,
+        payload: response,
+      });
+    } catch (err) {
+      dispatch({
+        type: SETTINGS_ACTION_TYPES.FETCH_SETTINGS_FAILURE,
+        payload: {
+          message: {
+            body: err.message,
+            title: __('Unable to find settings data', 'web-stories'),
+          },
+        },
+      });
+    }
+  }, [dataAdapter, globalStoriesSettingsApi]);
+
+  const updateSettings = useCallback(
+    async (settings) => {
+      const query = { googleAnalyticsId: settings.googleAnalyticsId };
+
+      try {
+        const response = await dataAdapter.post(
+          queryString.stringifyUrl({
+            url: globalStoriesSettingsApi,
+            query,
```

Comment:
Super nit, it seems like you could just pass ing the `settings` object (and not create the query object), this way this method won't have to be updated if/when new settings properties get created.  But, only a super nit, it's totally fine as is :)

Reply:
That would mean we have to name everything the same way that the query expects it which might be annoying. But that said, it's kinda moot right now since this isn't actually doing anything 🙃 I'll update it to keep it simpler!

Reply:
You make a good point, I forgot we may be translating javascript variable syntax to wordpress variable syntax.  At first I saw the settings object more like a story object, where you just pass the whole thing back and forth, and that may be the case in the long run, but it also might not be once things get setup and the API is working 😱

Reply:
we'll find out! hah! for the time being, i think this was a good thing to bring up to keep the structure as simple and un-opinionated as possible!